syft/frameworks/torch/tensors/interpreters/promise.py

```diff
@@ -0,0 +1,200 @@
+import syft as sy
+from syft.workers.abstract import AbstractWorker
+import weakref
+
+from syft.generic.tensor import AbstractTensor
+from syft.generic.tensor import initialize_tensor
+from syft.messaging.promise import Promise
+from syft.generic.frameworks.hook import hook_args
+
+
+class PromiseTensor(AbstractTensor, Promise):
+    def __init__(
+        self,
+        shape,
+        owner=None,
+        id=None,
+        tags=None,
+        description=None,
+        tensor_id=None,
+        tensor_type=None,
+        plans=None,
+    ):
+        """Initializes a PromiseTensor
+
+        Args:
+            owner: An optional BaseWorker object to specify the worker on which
+                the tensor is located.
+            id: An optional string or integer id of the LoggingTensor.
+        """
+
+        if owner is None:
+            owner = sy.local_worker
+
+        # I did check that the two __init__ methods below only get called once, but it
+        # was exhibiting some strange behavior when I used super() for both of them.
+
+        # constructor for AbstractTensor
+        super().__init__(id=id, owner=owner, tags=tags, description=description)
+
+        # constructor for Promise
+        Promise.__init__(self, owner=owner, obj_type=tensor_type, plans=plans)
+
+        self._shape = shape
+
+        del self.child
+
+    def torch_type(self):
+        return self.obj_type
+
+    @property
+    def shape(self):
+        return self._shape
+
+    @property
+    def grad(self):
+        return None
+        # if not hasattr(self, "_grad"):
+        #     self._grad = PromiseTensor(shape=self._shape, tensor_type=self.torch_type()).wrap()
+        #
+        # return self._grad
+
+    def on(self, tensor: "AbstractTensor", wrap: bool = True) -> "AbstractTensor":
```

Comment:
That's the same piece of code than for AbstractTensor? If so we can remove it :)

Reply:
This line is added compared to AbstractTensor:
`self.obj_type = tensor.type()`

Reply:
Could you add that line then call super.on to avoid rewriting the same code

Reply:
Would be a good idea!
Can someone just remind me what .on() is used for? I'm not sure I understand when it would be needed for promises (actually, it's never used at this point). If it's never useful, I'll remove that

Reply:
That's what I thought but in this case, I don't think we would ever need it in the case of Promises, am I wrong?
I'll remove it